embulk-core/src/main/java/org/embulk/jruby/JRubyScriptingModule.java

```diff
@@ -176,56 +173,31 @@ public ScriptingContainerDelegate get() throws ProvisionException
                 jruby.remove("__internal_classpath__");
             }
 
-            // Search embulk/java/bootstrap.rb from a $LOAD_PATH.
-            // $LOAD_PATH is set by lib/embulk/command/embulk_run.rb if Embulk starts
-            // using embulk-cli but it's not set if Embulk is embedded in an application.
-            // Here adds this jar's internal resources to $LOAD_PATH for those applciations.
-
-//            List<String> loadPaths = new ArrayList<String>(jruby.getLoadPaths());
-//            String coreJarPath = JRubyScriptingModule.class.getProtectionDomain().getCodeSource().getLocation().getPath();
-//            if (!loadPaths.contains(coreJarPath)) {
-//                loadPaths.add(coreJarPath);
-//            }
-//            jruby.setLoadPaths(loadPaths);
-
-            // load embulk.rb
-            jruby.runScriptlet("require 'embulk'");
-
-            // jruby searches embulk/java/bootstrap.rb from the beginning of $LOAD_PATH.
+            // Embulk's base Ruby code is loaded at last.
+            jruby.runScriptlet("require 'embulk/logger'");
             jruby.runScriptlet("require 'embulk/java/bootstrap'");
 
-            // TODO validate Embulk::Java::Injected::Injector doesn't exist? If it already exists,
-            //      Injector is created more than once in this JVM although use_global_ruby_runtime
-            //      is set to true.
-
-            // set some constants
-            jruby.callMethod(
-                    jruby.runScriptlet("Embulk::Java::Injected"),
-                    "const_set", "Injector", injector);
-            jruby.callMethod(
-                    jruby.runScriptlet("Embulk::Java::Injected"),
-                    "const_set", "ModelManager", injector.getInstance(ModelManager.class));
-            jruby.callMethod(
-                    jruby.runScriptlet("Embulk::Java::Injected"),
-                    "const_set", "BufferAllocator", injector.getInstance(BufferAllocator.class));
-
-            // initialize logger
-            jruby.callMethod(
-                    jruby.runScriptlet("Embulk"),
-                    "logger=",
-                        jruby.callMethod(
-                            jruby.runScriptlet("Embulk::Logger"),
-                            "new", injector.getInstance(ILoggerFactory.class).getLogger("ruby")));
+            final Object injected = jruby.runScriptlet("Embulk::Java::Injected");
+            jruby.callMethod(injected, "const_set", "Injector", injector);
+            jruby.callMethod(injected, "const_set", "ModelManager", injector.getInstance(ModelManager.class));
+            jruby.callMethod(injected, "const_set", "BufferAllocator", injector.getInstance(BufferAllocator.class));
```

Comment:
I'm still not sure about why `jruby.runScriptlet("Embulk::Java::Injected")` is evaluated 3 times but, I think that the reusing should be ok.